### PR TITLE
Bump to Go 1.18

### PR DESCRIPTION
**Pull request by realshuting**
Signed-off-by: ShutingZhao <shuting@nirmata.com>

## Related Issue(s)

Closes https://github.com/kyverno/policies/issues/387.

## Description

This PR bumps to Go 1.18.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Comment by chipzoller** [comment]
Already fixed in https://github.com/kyverno/policies/pull/388

**Comment by realshuting** [comment]
I don't see it on the `release-1.8` branch.

**Comment by chipzoller** [comment]
Shoot, sorry, I didn't see this was for a release branch.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,9 +28,9 @@ jobs:
           ref: ${{ github.base_ref || github.ref_name }}
 
       - name: Set up Go 
-        uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f
+        uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4 # v2.1.5
```
Comment (chipzoller) [inline .github/workflows/ci.yml:31]: We don't want to go back that far, we want latest of 3.3.0 (hash `268d8c0ca0432bb2cf416faae41297df9d262d7f`)

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (realshuting) [inline .github/workflows/ci.yml:31]:
```suggestion
        uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f # v2.1.5
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,9 +28,9 @@ jobs:
           ref: ${{ github.base_ref || github.ref_name }}
 
       - name: Set up Go 
-        uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f
+        uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f # v2.1.5
```
Comment (realshuting) [inline .github/workflows/ci.yml:31]:
```suggestion
        uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f
```

**Review by chipzoller — APPROVED** [review]

**Pull request merged**